test/convergence/test_mini_models.py

```diff
@@ -29,6 +29,13 @@
     apply_liger_kernel_to_mixtral,
```

Comment:
is it possible to just separate bf16 and fp32 to two pytest funcs? 

```python
def _test_convergence_once(...):
  ...

def test_convergence_fp32(...):
  ...

def test_convergence_bf16(...):
  ...
```

I am not sure if these two pytest func will share the same dependencies. else, we can just move revert_* to testing utils.

Reply:
@ByronHsu smart move! understood, thanks a lot!

It's much cleaner with fewer public-facing code changes. I'll remove the `revert_*` items. But I guess it won't work for `mini_gemma1.1` with `with_liger` disabled case as `gemma1` and `gemma1.1` still share the same module namespace `transformers.models.gemma.modeling_gemma`.

I also think it would be helpful to add a caveat, advising users to reload the modules themselves if necessary, to prevent similar 'test against itself' issues from occurring again.

Reply:
It's sad that separating `test_convergence` is not working. I'll go ahead for move `revert_*` to testing utils

Reply:
FYI, a naive example (thanks to gpt):

```python
# patch_module.py

def patched_function():
    return "patched"

def apply_patch():
    # Monkey patch the function globally by importing the original module
    import test_module
    test_module.original_function = patched_function

def remove_patch():
    # Restore the original function
    import test_module
    test_module.original_function = original_function

# Save the original function to restore later
from test_module import original_function

```

```python
# test_module.py

def original_function():
    return "original"

def test_function_1():
    # Import patch_module to apply the patch
    import patch_module

    # Apply the patch
    patch_module.apply_patch()

    # Call the patched function
    assert original_function() == "patched"

def test_function_2():
    # Ensure original function is restored 
    assert original_function() == "original"

```

```shell
⚡ ~ pytest ./test_module.py
============================================================================ test session starts ============================================================================
platform linux -- Python 3.10.10, pytest-8.3.2, pluggy-1.5.0
rootdir: /teamspace/studios/this_studio
plugins: anyio-4.4.0
collected 2 items                                                                                                                                                           

t/test_module.py .F                                                                                                                                                   [100%]

================================================================================= FAILURES ==================================================================================
______________________________________________________________________________ test_function_2 ______________________________________________________________________________

    def test_function_2():
        # Ensure original function is restored
>       assert original_function() == "original"
E       AssertionError: assert 'patched' == 'original'
E         
E         - original
E         + patched

t/test_module.py:18: AssertionError
========================================================================== short test summary info ==========================================================================
FAILED t/test_module.py::test_function_2 - AssertionError: assert 'patched' == 'original'
======================================================================== 1 failed, 1 passed in 0.05s ========================================================================
⚡ ~ 
```